src/PortingEmbed/PortingFormContainer.tsx

```diff
@@ -73,18 +75,32 @@ export function PortingFormContainer({
     return subscription?.porting ? nextPortingStep(subscription?.porting) : null
   }, [subscription?.porting])
 
+  const portingIsCompleted =
+    subscription?.porting &&
+    hasAllRequiredFieldsCompleted(subscription.porting) &&
+    !subscription.porting.declinedCode
+
+  const portingIsDeclined =
+    subscription?.porting && subscription?.porting.status === 'declined'
```

Comment:
should we shorten it?

```suggestion
    subscription?.porting?.status === 'declined'
```